src/Indicator.vala

```
@@ -129,6 +137,26 @@ public class Session.Indicator : Wingpanel.Indicator {
             }
 
             main_grid.add (suspend);
+            
+            try {
+                if(suspend_interface.can_hibernate () == "yes"){
+                    main_grid.add (hibernate);
+                }
+            } catch (GLib.Error e) {
+                stderr.printf ("%s\n", e.message);
```

This seems like it should throw an error dialog and not fail silently